### DISCUSSION TITLE
Modeling of composite material

### DISCUSSION THREAD

**Discussion by HoseongJeong** [kinnala/scikit-fem#1168] [Q&A] [answered]
Hello, scikit-fem users and developers,

I am using scikit-fem to perform analysis on civil engineering structures.

In our field, we typically model circular reinforcement bars within a rectangular concrete block for FEM analysis. In this case, how should we generate the mesh and basis? (It is assumed that the two objects are perfectly bonded.)

Additionally, since there are two types of materials, I have prepared two bilinear forms and linear forms. How should I construct the matrix in this situation?

Thank you for your great help.

**Comment by kinnala** [comment, marked as answer]
The best option is to create a mesh with nodes at the interface of the two materials, and tag the elements into two subdomains ’concrete’ and ’steel’. Then you can initialize two Basis objects as Basis(m, e, elements=’concrete’) and Basis(m, e, elements=’steel’) and assemble two matrices. Finally, you can sum the two matrices to get the stiffness matrix for the whole structure.
